### PR TITLE
Allowlist Sample

### PR DESCRIPTION
**Pull request by yschimke**
An Android Sample for allowing requests to a specific dev server e.g. localhost.  All other traffic is still validated.

### CODE REVIEW

**Comment by yschimke** [comment]
Not sure how to test on JVM :(

```
Skipping all security checks
[CN=localhost]
Skipping all security checks
[CN=www.google.com, O=Google LLC, L=Mountain View, ST=California, C=US, CN=GTS CA 1O1, O=Google Trust Services, C=US]
https://www.google.com/robots.txt
```

**Comment by yschimke** [comment]
Going to test with X509ExtendedTrustManager

File: samples/guide/src/main/java/okhttp3/recipes/kt/DevServerJVM.kt [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      delegate.checkServerTrusted(chain, authType, engine)
+    }
+  }
+
+  fun isAllowed(host: String): Boolean = hosts.contains(host)
+
+  override fun getAcceptedIssuers(): Array<X509Certificate> = delegate.acceptedIssuers
+}
+
+class DevServerJVM {
```
Comment (swankjesse) [inline samples/guide/src/main/java/okhttp3/recipes/kt/DevServerJVM.kt]: supernit: Jvm

**Comment by yschimke** [comment]
This was the point of the issue I raised. What is the API like?

**Comment by yschimke** [comment]
@swankjesse Going to land as it is for now, the work out what the nice public API for this would be.  I don't think it's trivial.

**Pull request merged**